find-gas-usage.js

```diff
@@ -0,0 +1,121 @@
+// usage: tweak constants at head of file, then run "node find-gas-usage.js"
+//        for very common events/calls, you may need to use a smaller TO-FROM
+//        range to keep web3 from yelling at you
+
+const ORIGIN = 6784800; // launch ceremony begins
+const PUBLIC = 7033765; // azimuth becomes public
+const LATEST = 'latest';
+
+const FROM = ORIGIN;
+const TO = LATEST;
+
+const VENT = 'ChangedKeys';
+const FUNC = '0x4447e48c'; // configureKeys
+const ARGS = 5;
+
+// const VENT = 'Spawned';
+// const FUNC = '0xa0d3253f'; // spawn
+// const ARGS = 2;
+
+// const VENT = 'OwnerChanged';
+// const FUNC = '0x1e79a85b'; // transferPoint
+// const ARGS = 3;
+
+// const VENT = 'ChangedSpawnProxy';
+// const FUNC = '0xae326221'; // setSpawnProxy
+// const ARGS = 2;
+
+// const VENT = 'ChangedManagementProxy';
+// const FUNC = '0x8866bb2c'; // setManagementProxy
+// const ARGS = 2;
+
+// const VENT = 'ChangedTransferProxy';
+// const FUNC = '0x2c7ba564'; // setTransferProxy
+// const ARGS = 2;
+
+// const VENT = 'ChangedVotingProxy';
+// const FUNC = '0xa60e8bd6'; // setVotingProxy
+// const ARGS = 2;
+
+// const VENT = 'EscapeRequested';
+// const FUNC = '0xbf5772b9'; // escape
+// const ARGS = 2;
+
+// const VENT = 'EscapeCanceled';
+// const FUNC = '0xc6d761d4'; // cancelEscape
+// const ARGS = 1;
+
+// const VENT = 'EscapeAccepted';
+// const FUNC = '0xc1b9d98b'; // adopt
+// const ARGS = 1;
+
+// const VENT = 'EscapeCanceled';
+// const FUNC = '0xbbe21ca5'; // reject
+// const ARGS = 1;
+
+// const VENT = 'LostSponsor';
+// const FUNC = '0x073a7804'; // detach
+// const ARGS = 1;
+
+const LENT = 64 * ARGS + FUNC.length;
+
+const Web3 = require('web3');
+const azimuth = require('azimuth-js');
+
+(async () => {
+  const web3 = new Web3(
+    new Web3.providers.HttpProvider(
+      'https://mainnet.infura.io/v3/196a7f37c7d54211b4a07904ec73ad87'
```

Comment:
this API key needs to be an env variable, and we should probably cycle it now that it's been public on github

Reply:
It's not a secret, we include it in the urbit source code also. It's just an endpoint URL. Fair on making is a constant though!

Reply:
ah, cool 👍